frontend/src/metabase-lib/lib/metadata/Database.ts

```diff
@@ -137,42 +183,12 @@ export default class Database extends Base {
     });
   }
 
-  nativeQuery(native) {
-    return this.nativeQuestion(native).query();
-  }
-
   /** Returns a database containing only the saved questions from the same database, if any */
-  savedQuestionsDatabase() {
-    return this.metadata.databasesList().find(db => db.is_saved_questions);
+  savedQuestionsDatabase(): Database | undefined {
+    return this.metadata?.databasesList().find(db => db.is_saved_questions);
```

Comment:
In other places like `schema(...)` you've added `?? null`. Should we also make it similar here so that the return type is `Database | null`?